<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>StdOtauWidget</class>
 <widget class="QWidget" name="StdOtauWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>518</width>
    <height>561</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OTA Update</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>OTAU Update</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4"/>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string>Otau File</string>
           </property>
           <property name="buddy">
            <cstring>ou_fileEdit</cstring>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="ou_fileEdit">
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="ou_fileSelectButton">
           <property name="maximumSize">
            <size>
             <width>32</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Chose otau file for selected node.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="0" column="0">
          <widget class="QLabel" name="label_11">
           <property name="text">
            <string>Manufacturer Id</string>
           </property>
           <property name="buddy">
            <cstring>ou_manufacturerEdit</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="ou_manufacturerEdit">
           <property name="inputMask">
            <string>\0\xHHHHHHHH</string>
           </property>
           <property name="text">
            <string>0x0000</string>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Image Type</string>
           </property>
           <property name="buddy">
            <cstring>ou_imageTypeEdit</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="3" colspan="2">
          <widget class="QLineEdit" name="ou_imageTypeEdit">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The otau client might use the image type as additional filter to accept or deny a otau image.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="inputMask">
            <string>\0\xHHHHHHHH</string>
           </property>
           <property name="text">
            <string>0x0000</string>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>File Version</string>
           </property>
           <property name="buddy">
            <cstring>ou_fileVersionEdit</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="ou_fileVersionEdit">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="inputMask">
            <string>\0\xHHHHHHHH</string>
           </property>
           <property name="text">
            <string>0x00000000</string>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QLabel" name="label_12">
           <property name="text">
            <string>Size</string>
           </property>
           <property name="buddy">
            <cstring>ou_SizeEdit</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="3" colspan="2">
          <widget class="QLineEdit" name="ou_SizeEdit">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Size of OTAU file including header.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>0x00000000</string>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QCheckBox" name="restartAfterUpgradeCheckbox">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Restart After Upgrade</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_13">
           <property name="text">
            <string>Last Image Query</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QLabel" name="lastQueryLabel">
           <property name="text">
            <string>00:00:00</string>
           </property>
          </widget>
         </item>
         <item row="2" column="4">
          <widget class="QCheckBox" name="useAcksCheckBox">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>Use ACKs</string>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="3" column="4">
          <widget class="QCheckBox" name="usePageRequestCheckBox">
           <property name="text">
            <string>Page Request</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QSpinBox" name="restartAfterUpgradeSpinBox">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="suffix">
            <string> s</string>
           </property>
           <property name="maximum">
            <number>3600</number>
           </property>
           <property name="value">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QSpinBox" name="spacingSpinBox">
           <property name="suffix">
            <string> ms</string>
           </property>
           <property name="prefix">
            <string/>
           </property>
           <property name="minimum">
            <number>4</number>
           </property>
           <property name="maximum">
            <number>8000</number>
           </property>
           <property name="value">
            <number>100</number>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_14">
           <property name="text">
            <string>Packet Spacing</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="labelOtauState">
           <property name="text">
            <string>OTAU enabled</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::MinimumExpanding</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="ou_queryButton">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Requests the selected node to query for new update image.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Query</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="ou_abortButton">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Aborts update for selected node.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Abort</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="ou_updateButton">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Starts update for selected node. File must be chosen first.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Update</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QTableView" name="tableView">
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectRows</enum>
         </property>
         <attribute name="horizontalHeaderStretchLastSection">
          <bool>true</bool>
         </attribute>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>OTAU File</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QLabel" name="fileNameLabel">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QFormLayout" name="formLayout_2">
         <item row="0" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>File Version</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="of_FileVersionEdit">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Version field is a 32-bit value 0xXYBBPPRR&lt;/p&gt;&lt;p&gt;X - Major release&lt;/p&gt;&lt;p&gt;Y - Minor release&lt;/p&gt;&lt;p&gt;BB - Application build&lt;/p&gt;&lt;p&gt;PP - Platform&lt;/p&gt;&lt;p&gt;RR - Reserved (Set to 0)&lt;/p&gt;&lt;p&gt;Example: 0x14420100 Version 1.4 build 42 Platform 0x01&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>0x00000000</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Header Version</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="of_headerVersionEdit">
           <property name="text">
            <string>0x0100</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>Image Type</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLineEdit" name="of_imageTypeEdit">
           <property name="text">
            <string>0x0000</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Manufacturer Code</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QLineEdit" name="of_manufacturerEdit">
           <property name="text">
            <string>0x1135</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>ZigBee Stack Version</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QLineEdit" name="of_zigbeeStackVersionEdit">
           <property name="text">
            <string>0x0002</string>
           </property>
          </widget>
         </item>
         <item row="6" column="0">
          <widget class="QLabel" name="label_9">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>Description</string>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QPlainTextEdit" name="of_descriptionEdit">
           <property name="enabled">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="10" column="0">
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>Raw Firmware Size</string>
           </property>
          </widget>
         </item>
         <item row="10" column="1">
          <widget class="QLineEdit" name="of_firmwareSizeEdit">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Size of raw firmware (without OTAU file header).&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>0x1FFFF</string>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="7" column="0">
          <widget class="QLabel" name="label_15">
           <property name="text">
            <string>Min HW Version</string>
           </property>
          </widget>
         </item>
         <item row="7" column="1">
          <widget class="QLineEdit" name="of_minHwVersionEdit">
           <property name="text">
            <string>0x0000</string>
           </property>
          </widget>
         </item>
         <item row="8" column="0">
          <widget class="QLabel" name="label_16">
           <property name="text">
            <string>Max HW Version</string>
           </property>
          </widget>
         </item>
         <item row="8" column="1">
          <widget class="QLineEdit" name="of_maxHwVersionEdit">
           <property name="text">
            <string>0x0000</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QPushButton" name="saveButton">
           <property name="text">
            <string>Save</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="saveAsButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Save As</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="openButton">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Open a *.zigbee file&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Open</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>restartAfterUpgradeCheckbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>restartAfterUpgradeSpinBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel">
     <x>201</x>
     <y>133</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
